packages/ibm-products/src/components/ConditionBuilder/ConditionBuilderItem/ConditionBuilderItem.js

```diff
@@ -24,15 +24,26 @@ export const ConditionBuilderItem = ({
   ...rest
 }) => {
   const contentRef = useRef(null);
-  const [propertyLabel, setPropertyLabel] = useState(label);
+  const [propertyLabel, setPropertyLabel] = useState();
   const [open, setOpen] = useState(false);
+  const [isInvalid, setIsInvalid] = useState(false);
 
   useEffect(() => {
-    const propertyId =
-      rest['data-name'] == 'valueField' && type
-        ? valueRenderers[type](label, config)
-        : label;
-    setPropertyLabel(translateWithId(propertyId));
+    if (label == 'INVALID') {
```

Comment:
the `useEffect` here is unnecessary. `propertyLabel` and `invalid` are derived directly from props, so you can just define them as regular variables. something like
```
const propertyLabel = getPropertyLabel();
const isInvalid = label === 'INVALID' || label === 'DUPLICATE';
```

Reply:
modified. Good point. :)